icaruscode/Decode/MCDecoderICARUSTPCwROI_module.cc

```diff
@@ -498,110 +499,54 @@ void MCDecoderICARUSTPCwROI::processSingleLabel(art::Event&
-                }
-
-                boardMapItr = mapItr;
-                boardWireCountMap[readoutBoardID] = 0;
-            }
-
-           // Decompress data into local holder
-            raw::Uncompress(rawDigit->ADCs(), rawDataVec, rawDigit->Compression());
-
-            // Fill into the data structure
-            icarus_signal_processing::VectorFloat& boardDataVec = boardMapItr->second.second[wireIdx];
-
-            for(size_t tick = 0; tick < dataSize; tick++) boardDataVec[tick] = rawDataVec[tick];
-
-            boardMapItr->second.first[wireIdx] = daq::INoiseFilter::ChannelPlanePair(channel,planeIdx);
-
-            if (++boardWireCountMap[readoutBoardID] == MAXCHANNELS)
-            {
-                processSingleImage(clockData, boardMapItr->second, coherentNoiseGrouping, concurrentRawDigits, concurrentRawRawDigits, coherentRawDigits, concurrentROIs);
-
-                boardToChannelArrayPairMap.erase(boardMapItr);
-
-                // Get the wireIDs for this channel
-//                std::vector<geo::WireID> wids = fGeometry->ChannelToWire(channel);
-//
-//                // Since we work with channels we can ignore the case in the middle of the TPC where there are 
-//                // wires crossing the midplane and just work with the first wire ID
-//                const geo::WireID&  wireID =  wids[0];
-//                const geo::PlaneID& planeID = wireID.planeID();
-//
-//                // Ok, now store things...
-//                unsigned int planeIndex = fPlaneToROPPlaneMap.find(planeID)->second;
-//                unsigned int wire       = channel - fPlaneToWireOffsetMap.find(planeID)->second;
-//
-//                if (wire >= channelArrayPairVec[planeIndex].second.size()) continue;
-//
-//                icarus_signal_processing::VectorFloat& dataVec = channelArrayPairVec[planeIndex].second[wire];
-//
-//                for(size_t tick = 0; tick < dataSize; tick++) dataVec[tick] = rawDataVec[tick];
-//
-//                // Keep track of the channel
-//                channelArrayPairVec[planeIndex].first[wire] = daq::INoiseFilter::ChannelPlanePair(channel,planeID.Plane);
-            }
-        }
-
-        // Some detector simulations don't output channels that don't have any possibility of signal (ghost channels)
-        // Do a cleanup phase here to find these
-        for(auto& boardInfo : boardToChannelArrayPairMap)
-        {
-            if (boardWireCountMap[boardInfo.first] < 64)
-            {
-                processSingleImage(clockData, boardInfo.second, boardWireCountMap[boardInfo.first], concurrentRawDigits, concurrentRawRawDigits, coherentRawDigits, concurrentROIs);
-            }
-        }
+	  unsigned int board = channelToBoardItr->second.first;
+	  auto mapIter = boardToRawDigitMap.find(board);
+	  if (mapIter != boardToRawDigitMap.end()) {
+	    mapIter->second.push_back(rawDigit);
+	  } else {
+	    boardToRawDigitMap.insert({board,std::vector<const raw::RawDigit*>{rawDigit}});
```

Comment:
This should be equivalent to:
```suggestion
          boardToRawDigitMap.try_emplace(board).first->push_back(&rawDigit);
```
(also added address operator `&` for my previous suggestion).
You can argue whether it is an improvement or not (`first` is quite cryptic...).

Reply:
yeah, why first? I would have thought second as in line 519

Reply:
actually it should be `boardToRawDigitMap.try_emplace(board).first->second.push_back(&rawDigit);` and I am also not sure it is an improvement

Reply:
so I am not going to make this change, but it can be done if you have a strong opinion